fatresize.c

```diff
@@ -621,6 +621,11 @@ int main(int argc, char **argv) {
     opts.size = constraint->max_size * dev->sector_size;
     ped_constraint_destroy(constraint);
   }
+  if (opts.size >= dev->sector_size * part_geom.length) {
```

Comment:
Currently fatresize has two thing that it does, one is obvious, resizing a FAT filesystem, and the other less obvious, resizing the partition on whch a FAT filesystem resides. This would effectively remove the capability to resize the partition, which I actually think would be a good thing.